### PR TITLE
fix(usenet): use reader probe for provider health

### PR DESCRIPTION
**Pull request by git-geeky**
## Problem

NNTP providers that support reader commands but not the optional `DATE` extension fail the dashboard connection test. The pool also releases a connection after a failed idle probe without closing its socket, which can leak provider connections and race queued work.

## Fix

Use a no-payload `STAT` request for authenticated reader health checks. While an idle probe owns a pool slot, queued work waits; failed probes destroy the socket before the slot is released and redialed.

Related: #1035

## Verification

- Added four fake-NNTP regressions covering reusable missing-article probes, authenticated dashboard checks, failed-probe close/redial, and delayed-probe work serialization.
- `pnpm -F @aiostreams/core build`
- `pnpm -F @aiostreams/server build`
- Verified the protocol path against a provider that returns `400` to `DATE` but continues to answer `STAT` on the same authenticated TLS connection.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Usenet connection health checks by validating reader access directly.
  * Prevented queued work from running during connection checks.
  * Failed health checks now close affected connections and automatically reconnect.
  * Health checks correctly handle servers reporting a requested article as unavailable.

### CODE REVIEW

File: packages/core/src/usenet/nntp/provider-worker-pool.ts [hunk 1]
```diff
@@ -574,19 +577,29 @@ export class ProviderWorkerPool {
...
+            // explicitly before releasing the slot so the socket cannot leak.
+            conn.destroy();
+            slot.conn = null;
+          }
         })
-        .finally(() => this.dispatch());
+        .finally(() => {
+          slot.probing = false;
+          this.dispatch();
+        });
```
[the remark this answers is not in the document]
Comment (git-geeky) [reply packages/core/src/usenet/nntp/provider-worker-pool.ts:609]: > Deduct the remaining pipeline capacity for every probing slot. Add a regression test that asserts `freeSlots` is zero while the only slot has a pending probe.

Addressed in 8a97129e. `freeSlots` now reserves the probing slot's full remaining pipeline capacity based on its actual in-flight count, and the pending-probe regression asserts that the only slot advertises zero free capacity. The four focused NNTP tests, core TypeScript build, and formatting checks pass.

File: packages/core/src/usenet/nntp/health-probe.test.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    let conn: NntpConnection | undefined;
+
+    try {
+      conn = await NntpConnection.connect(
+        provider(fake.port),
+        CONNECTION_OPTIONS
+      );
+      await conn.probeReader(undefined, 1_000);
+      assert.deepEqual(fake.commands, [HEALTHCHECK_COMMAND]);
+      assert.equal(conn.isUsable, true);
```
[the remark this answers is not in the document]
Comment (git-geeky) [reply packages/core/src/usenet/nntp/health-probe.test.ts:186]: Good catch. This is a valid gap: `isUsable` only confirms socket state, so the new `223`/`423` cases do not yet prove that the probe response left the FIFO protocol queue aligned.

I’ll extend the case table with the expected follow-up result, issue a second lightweight `STAT` through `conn.stat(...)`, and assert `true` after `223` and `false` after `423`, while retaining the `HEALTHCHECK_COMMAND` and `isUsable` assertions. I’m leaving this thread unresolved until that change is pushed and verified.

[the remark this answers is not in the document]
Comment (git-geeky) [reply packages/core/src/usenet/nntp/health-probe.test.ts:186]: > After `probeReader`, send an existing lightweight command and assert its response for both `223` and `423`.

Addressed in 2046616d. Each case now sends a second `STAT <second@test.invalid>`, asserts `true` for `223` and `false` for `423`, and verifies the two-command order while retaining the `HEALTHCHECK_COMMAND` and `isUsable` assertions. `pnpm -F @aiostreams/core test` passes 6/6; the core/server builds and Prettier check also pass.
